### PR TITLE
Create Select Component

### LINKED ISSUE #7: Create Select Component
image

- components/select
- Should have the dropdown capability when clicking on the filter
- Make sure to have props for the current option, placeholder, and options
- Checkout the hackathon website implementation of select for state management

### PR DESCRIPTION
**Pull request by khandrew1**
image
image
image

* Created Select component
* Uses React Bootstrap dropdown as a base
* Current option denoted with a dot
* Options are passed in via props

* Closes #7

### CODE REVIEW

**Comment by shahdivyank** [comment]
editable prop is not needed

**Comment by shahdivyank** [comment]
u should be able to do padding to get the height without hardcoding

**Comment by shahdivyank** [comment]
use client is chill

File: src/components/Select.jsx [hunk 1]
```diff
@@ -1,5 +1,57 @@
-const Select = () => {
-  return <div>Filter</div>;
+"use client";
+
+import { BiSolidUpArrow } from "react-icons/bi";
+import { BsDot } from "react-icons/bs";
+import Dropdown from "react-bootstrap/Dropdown";
+import { useState } from "react";
+
+const Toggle = ({ onClick, show, currentOption, placeholder }) => {
```
Comment (shahdivyank) [inline src/components/Select.jsx]: call it option or just current

File: src/components/Select.jsx [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+      <BiSolidUpArrow
+        className={`${
+          show && "rotate-180"
+        } duration-300 text-code-white text-xs`}
+      />
+    </button>
+  );
+};
+
+const Select = ({ currentOption, setCurrentOption, placeholder, options }) => {
```
Comment (shahdivyank) [inline src/components/Select.jsx]: call it just current or just option

File: src/components/Select.jsx [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+      <div className="p-1" />
+      <Dropdown.Menu className="!bg-code-darkgray w-1/6 pt-10">
+        {options.map((option, index) => (
+          <Dropdown.Item
+            className="hover:!bg-code-lightgray !bg-code-darkgray !text-code-white !flex !flex-row !justify-between !items-center"
+            key={index}
+            onClick={() => setCurrentOption(option)}
+          >
+            {option}
+            {option == currentOption && <BsDot className="text-3xl" />}
```
Comment (shahdivyank) [inline src/components/Select.jsx]: triple ===

File: src/components/Select.jsx [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+      onToggle={() => setShow(!show)}
+      autoClose={true}
+    >
+      <Dropdown.Toggle
+        show={show}
+        current={current}
+        placeholder={placeholder}
+        as={Toggle}
+      />
+      <div className="p-1" />
```
Comment (shahdivyank) [inline src/components/Select.jsx:40]: what's the point of this?

Comment (khandrew1) [reply src/components/Select.jsx:40]: the figma file has a pretty big gap between the button and menu, was just following that

**Review by shahdivyank — APPROVED** [review]

**Pull request merged**